samples/core/model_serving_component/model_serving_component.ipynb

```diff
@@ -0,0 +1,297 @@
+    "MODEL_VERSION = '1'                                            # A number representing the version model \n",
+    "OUTPUT_BUCKET = 'gs://%s-serving-component' % PROJECT_NAME     # A GCS bucket for asset outputs\n",
+    "KUBEFLOW_DEPLOYER_IMAGE = 'gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:fe639f41661d8e17fcda64ff8242127620b80ba0'\n",
+    "MODEL_PATH = '%s/%s' % (OUTPUT_BUCKET,MODEL_NAME)               \n",
+    "MODEL_VERION_PATH = '%s/%s/%s' % (OUTPUT_BUCKET,MODEL_NAME,MODEL_VERSION)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating gs://chavoshi-dev-2-serving-component/...\n"]
+    }
+   ],
+   "source": [
+    "!gsutil mb {OUTPUT_BUCKET}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Get or create an experiment and submit a pipeline run\n",
+    "import kfp\n",
+    "client = kfp.Client()\n",
+    "\n",
+    "try:\n",
+    "    experiment = client.get_experiment(experiment_name=EXPERIMENT_NAME)\n",
+    "except:\n",
+    "    experiment = client.create_experiment(EXPERIMENT_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load a Keras Model \n",
+    "Loading a pretrained Keras model to use as an example. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING: Logging before flag parsing goes to stderr.\n",
+      "W0813 22:25:44.771417 139829630277440 deprecation.py:506] From /opt/conda/lib/python3.6/site-packages/tensorflow/python/ops/init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
+      "Instructions for updating:\n",
+      "Call initializer instance with the dtype argument instead of passing it to the constructor\n"
+     ]
+    }
+   ],
+   "source": [
+    "model = tf.keras.applications.NASNetMobile(input_shape=None,\n",
+    "                                   include_top=True,\n",
+    "                                   weights='imagenet',\n",
+    "                                   input_tensor=None,\n",
+    "                                   pooling=None,\n",
+    "                                   classes=1000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Saved the Model for TF-Serve\n",
+    "Save the model using keras export_saved_model function. Note that specifically for TF-Serve the output directory should be structure as model_name/model_version/saved_model."
```

Comment:
Should this be bucket_name/model_name/model_version? What is the difference between saved_model and model_name here.

Reply:
The directory structure is the format that TF-serve expects. The model_name is the name of the root directory where the model versions are saved. Saved_model is the actual saved_model.pb file. (reference: https://www.tensorflow.org/tfx/serving/serving_basic)